evals/elsuite/ecn.py

```diff
@@ -0,0 +1,102 @@
+import random
+import re
+
+import evals
+import evals.metrics
+from evals import CompletionFn
+
+
+class Ecn(evals.Eval):
+    def __init__(
+        self,
+        completion_fns: list[CompletionFn],
+        samples_jsonl: str,
+        num_few_shot: int = 0,
+        few_shot_jsonl: str = None,
+        **kwargs
+    ):
+        super().__init__(completion_fns, **kwargs)
+        self.samples_jsonl = samples_jsonl
+        self.num_few_shot = num_few_shot
+        self.few_shot_jsonl = few_shot_jsonl
+
+        if self.num_few_shot > 0:
+            assert few_shot_jsonl is not None, "few shot requires few shot sample dataset"
+            self.few_shot_jsonl = few_shot_jsonl
+            self.few_shot = evals.get_jsonl(self.few_shot_jsonl)
+
+    def run(self, recorder):
+        """
+        Called by the `oaieval` CLI to run the eval. The `eval_all_samples` method calls `eval_sample`.
+        """
+
+        samples = self.get_samples()
+        self.eval_all_samples(recorder, samples)
+        events = recorder.get_events("match")
+        num_events = len(list(events))
+
+        total_score = sum(map(lambda e: e.data["score"], events))
+        average_score = total_score / num_events
+        valid_format = f"{round(sum(map(lambda e: e.data['valid_format'], events)) / num_events * 100, 2)}%"
+
+        return {
```

Comment:
How does this get spit out? I feel like I'm missing one last step in your notebook.

Reply:
It's in the logs?

Reply:
yes, it gets put in a log file in /tmp